lib/if.c

```diff
@@ -109,26 +112,25 @@ int if_cmp_name_func(char *p1, char *p2)
 	return 0;
 }
 
-static int if_cmp_func(struct interface *ifp1, struct interface *ifp2)
+static int if_cmp_func(const struct interface *ifp1,
+		       const struct interface *ifp2)
 {
-	return if_cmp_name_func(ifp1->name, ifp2->name);
+	return if_cmp_name_func((char *)ifp1->name, (char *)ifp2->name);
```

Comment:
Could you instead change the signature of if_cmp_name_func() ?

Reply:
I tried but 

```

Reply:
Once I find some time I'll try to rewrite `if_cmp_name_func()` to have const arguments. In the meanwhile I don't think this should block this PR.

Reply:
Yes I agree this is clearly another topic